ion/src/test/java/com/fasterxml/jackson/dataformat/ion/ionvalue/IonValueDeserializerTest.java

```diff
@@ -0,0 +1,171 @@
+    }
+
+    @Test
+    public void shouldBeAbleToDeserializeIncludingNullList() throws Exception {
+        IonValue ion = ion("{a:1, b:2, c:null.list}");
+
+        IonValueData data = ION_VALUE_MAPPER.parse(ion, IonValueData.class);
+
+        assertEquals(3, data.getAllData().size());
+        assertEquals(ion("1"), data.getAllData().get("a"));
+        assertEquals(ion("2"), data.getAllData().get("b"));
+        assertEquals(ion("null.list"), data.getAllData().get("c"));
+    }
+
+    @Test
+    public void shouldBeAbleToDeserializeNullList() throws Exception {
+        IonValue ion = ion("{c:null.list}");
+
+        IonValueData data = ION_VALUE_MAPPER.parse(ion, IonValueData.class);
+
+        assertEquals(1, data.getAllData().size());
+        assertEquals(SYSTEM.newNullList(), data.getAllData().get("c"));
+    }
+
+    @Test
+    public void shouldBeAbleToDeserializeNullStruct() throws Exception {
+        IonValue ion = ion("{c:null.struct}");
+
+        IonValueData data = ION_VALUE_MAPPER.parse(ion, IonValueData.class);
+
+        assertEquals(1, data.getAllData().size());
+        assertEquals(SYSTEM.newNullStruct(), data.getAllData().get("c"));
+    }
+
+    @Test
+    public void shouldBeAbleToDeserializeNullValue() throws Exception {
+        IonValue ion = SYSTEM.newNull();
+
+        IonValue data = ION_VALUE_MAPPER.parse(ion, IonValue.class);
+
+        assertEquals(ion, data);
+    }
+
+    @Test
+    public void shouldBeAbleToDeserializeAnnotatedNullStruct() throws Exception {
+        IonValue ion = ion("foo::null.struct");
+
+        IonValue data = ION_VALUE_MAPPER.parse(ion, IonValue.class);
+
+        assertEquals(ion, data);
+        assertEquals(1, data.getTypeAnnotations().length);
+        assertEquals("foo", data.getTypeAnnotations()[0]);
+    }
+
+    @Test
+    public void shouldBeAbleToDeserializeAnnotatedNullList() throws Exception {
+        IonValue ion = ion("foo::null.list");
+
+        IonValue data = ION_VALUE_MAPPER.parse(ion, IonValue.class);
+
+        assertEquals(ion, data);
+        assertEquals(1, data.getTypeAnnotations().length);
+        assertEquals("foo", data.getTypeAnnotations()[0]);
+    }
+
+    @Test
+    public void shouldBeAbleToSerializeAndDeserializePojo() throws Exception {
+        IonValueData source = new IonValueData();
+        source.put("a", ion("1"));
+        source.put("c", ion("null.list"));
+
+        IonValue data = ION_VALUE_MAPPER.serialize(source);
+        IonValueData result = ION_VALUE_MAPPER.parse(data, IonValueData.class);
+
+        assertEquals(source, result);
+    }
+
+    @Test
+    public void shouldBeAbleToSerializeAndDeserializeStringData() throws Exception {
+        StringData source = new StringData();
+        source.put("a", "1");
+        source.put("b", null);
+
+        IonValue data = ION_VALUE_MAPPER.serialize(source);
+        StringData result = ION_VALUE_MAPPER.parse(data, StringData.class);
+
+        assertEquals(source, result);
+    }
```

Comment:
Is this a sanity test to ensure that non-IonValue deserialization is not impacted?

Reply:
Yes. I've included some test cases not specifically covering the change but the entire `IonValueDeserializer` class since I was adding a new test class and added those as a kind of smoke/regression test.